### ISSUE TITLE
Context provider by folder fails with *

### ISSUE THREAD

**Issue by Mte90** [TabbyML/tabby#3437]
With the latest version I tried to scan a folder but is not clear if reads also the various subfolders.

I wrote as path `file:///[...]/Amber/*` (the [...] part is to censor) I got `crates/tabby-index/src/code/repository.rs:39: Failed to clone `file:///home/mte90/Desktop/kde/Amber/*`. Please check your repository configuration` in the terminal but on the UI there is no alert if it was fine or not.
I just did that because when using the chat panel was saying that just used the file on the root folder and not all the files in the various subfolders in that folder.

![image](https://github.com/user-attachments/assets/99241186-6a9a-460b-b190-d4fea1099428)

This is an example with a ffolder where there is a full django project where says `Read 8 files` instead of all of them.

**Comment by wsxiaoys** [comment]
Hi - repository path doesn't really support glob patterns, so you might wanna set `file:///[...]/Amber` as path.

**Comment by Mte90** [comment]
Yeah I saw that after trying but there is no warning/check in the Tabby UI. Like a validation about what I wrote, like if the path exists. To discover that you need to save and wait for the job to fails but you don't get any alert about it too.

**Comment by Mte90** [comment]
So about the second issue, the list include the files used to reply not the files scanned.

**Comment by wsxiaoys** [comment]
The Answer Engine will select only the chunks that are considered relevant to the question, rather than reading the entire content of the repository.

In future we might add functionality allowing user to explicitly including a file / directory in answer engine.
